src/main/java/com/adobe/abp/regola/rules/NumberRule.java

```diff
@@ -47,6 +47,10 @@ public NumberRule(Executor executor, String key, Operator operator, V value) {
 
     @Override
     Result check(V fact) {
+        if (fact == null || getValue() == null) {
+            return Result.INVALID;
```

Comment:
Return early if fact or value is missing

Reply:
This is already covered by:

```java
<F> Result checkFact(F fact, BiPredicate<F, T> predicate) {
        return Optional.ofNullable(fact)
                .filter(f -> Objects.nonNull(getValue()))
```